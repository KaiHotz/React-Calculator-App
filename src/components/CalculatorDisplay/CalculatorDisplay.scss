.calculator-display {
  color: white;
  background: #1c191c;
  line-height: 130px;
  font-size: 6em;
  position: relative;
  flex: 1;

  &__auto-scaling {
    display: inline-block;
    padding: 0 30px;
    position: absolute;
    right: 0;
    transform-origin: right;
  }
}
