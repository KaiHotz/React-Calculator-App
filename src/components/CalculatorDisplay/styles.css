.calculator-display {
  color: white;
  background: #1c191c;
  line-height: 130px;
  font-size: 6em;
  position: relative;
  flex: 1;
}
