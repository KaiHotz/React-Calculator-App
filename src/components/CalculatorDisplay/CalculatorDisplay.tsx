import { FC, useRef, useState, useEffect } from 'react';
import { getFormattedValue } from '../../utils/helpers';
import './CalculatorDisplay.scss';

export interface ICalculatorDisplayProps {
  value: string;
}

export const CalculatorDisplay: FC<ICalculatorDisplayProps> = ({ value = '0' }) => {
  const [scale, setScale] = useState<number>(1);
  const parentRef = useRef<HTMLDivElement>(null);
  const innerRef = useRef<HTMLDivElement>(null);

  // eslint-disable-next-line react-hooks/exhaustive-deps
  useEffect(() => {
    const availableWidth = parentRef?.current?.offsetWidth;
    const actualWidth = innerRef?.current?.offsetWidth;
    const actualScale = availableWidth && actualWidth ? availableWidth / actualWidth : 1;
    if (actualScale < 1) {
      setScale(actualScale);
    } else if (scale < 1) {
      setScale(1);
    }
  });

  return (
    <div className="calculator-display" ref={parentRef} data-testid="calculator-display">
      <div
        className="calculator-display__auto-scaling"
        style={{ transform: `scale(${scale},${scale})` }}
        ref={innerRef}
        data-testid="calculator-display-inner"
      >
        {getFormattedValue(value)}
      </div>
    </div>
  );
};
