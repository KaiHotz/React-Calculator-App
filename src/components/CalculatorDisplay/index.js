import React from 'react'
import PropTypes from 'prop-types'
import AutoScalingText from '../AutoScalingText'
import { getFormattedValue } from '../../utils/helper'
import './styles.css'

const CalculatorDisplay = ({ value }) => (
  <div className="calculator-display">
    <AutoScalingText>{getFormattedValue(value)}</AutoScalingText>
  </div>
)

CalculatorDisplay.propTypes = {
  value: PropTypes.string,
}

CalculatorDisplay.defaultProps = {
  value: '0',
}

export default CalculatorDisplay
