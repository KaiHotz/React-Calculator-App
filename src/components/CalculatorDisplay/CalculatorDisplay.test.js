import React from 'react'
import { shallow } from 'enzyme'
import {CalculatorDisplay} from './CalculatorDisplay'

describe('<CalculatorDisplay />', () => {
  it('should render', () => {
    const wrapper = shallow(<CalculatorDisplay />)

    expect(wrapper).toMatchSnapshot()
  })
})
