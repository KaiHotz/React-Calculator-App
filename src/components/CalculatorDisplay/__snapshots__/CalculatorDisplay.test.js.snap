// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalculatorDisplay /> should render 1`] = `ShallowWrapper {}`;
